datafu-spark/src/test/scala/datafu/spark/TestAggregators.scala

```diff
@@ -0,0 +1,265 @@
+    Seq(Row("a", 1, "asd1"),
+        Row("a", 2, "asd2"),
+        Row("a", 3, "asd3"),
+        Row("b", 1, "asd4")))
+
+  lazy val df =
+    sqlContext.createDataFrame(inputRDD, StructType(inputSchema)).cache
+
+  case class mapExp(map_col: Map[String, Int])
+  case class mapArrExp(map_col: Map[String, Array[String]])
+
+  lazy val defaultDbLocation = spark.sql("describe database default").toDF
+	.collect()
+	.filter(_.getString(0) == "Location")(0)(1)
+	.toString.replace("file:", "")
+
+  def deleteLeftoverFiles(table : String) : Unit = {
+   
+	val tablePath = Paths.get(defaultDbLocation + File.separator + table)
+	
+	// sanity check - only delete files if the path seems to be to a Hive warehouse
+	if (defaultDbLocation.endsWith("warehouse") && Files.exists(tablePath)) Files.walkFileTree(tablePath, new SimpleFileVisitor[Path] {
+      		override def visitFile(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        		Files.delete(path)
+        		FileVisitResult.CONTINUE
+      		}
+    	}
+  )
+ }
+
+  test("test multiset simple") {
+    val ms = udaf(new MultiSet())
+    val expected: DataFrame =
+      sqlContext.createDataFrame(List(mapExp(Map("b" -> 1, "a" -> 3))))
+    assertDataFrameEquals(expected, df.agg(ms($"col_grp").as("map_col")))
+  }
+
+  val mas = udaf(new MultiArraySet[String]())
+
+  test("test multiarrayset simple") {
+    assertDataFrameEquals(
+      sqlContext.createDataFrame(List(mapExp(Map("tre" -> 1, "asd" -> 2)))),
+      spark
+        .sql("select array('asd','tre','asd') arr")
+        .groupBy()
+        .agg(mas($"arr").as("map_col"))
+    )
+  }
+
+  test("test multiarrayset all nulls") {
+    // end case
+    spark.sql("drop table if exists mas_table")
+    deleteLeftoverFiles("mas_table")
+    	
+    spark.sql("create table mas_table (arr array<string>)")
+    spark.sql(
+      "insert overwrite table mas_table select case when 1=2 then array('asd') end " +
+        "from (select 1)z")
+    spark.sql(
+      "insert into table mas_table select case when 1=2 then array('asd') end from (select 1)z")
+    spark.sql(
+      "insert into table mas_table select case when 1=2 then array('asd') end from (select 1)z")
+    spark.sql(
+      "insert into table mas_table select case when 1=2 then array('asd') end from (select 1)z")
+    spark.sql(
+      "insert into table mas_table select case when 1=2 then array('asd') end from (select 1)z")
+
+    val expected = sqlContext.createDataFrame(List(mapExp(Map())))
+
+    val actual =
+      spark.table("mas_table").groupBy().agg(mas($"arr").as("map_col"))
+
+    assertDataFrameEquals(expected, actual)
+  }
+
+  test("test multiarrayset max keys") {
+    // max keys case
+    spark.sql("drop table if exists mas_table2")
+    deleteLeftoverFiles("mas_table2")
+
+    spark.sql("create table mas_table2 (arr array<string>)")
+    spark.sql(
+      "insert overwrite table mas_table2 select array('asd','dsa') from (select 1)z")
```

Comment:
Are all these 'z' aliases really needed?

Reply:
No, they're unnecessary - I took them from the old testing class ([TestSparkUDAFs](https://github.com/apache/datafu/pull/41/files#diff-884ddaa9627aad21382e15b67a113b045b2341b449bc5095af663961e923b1bc)). I will take advantage of this PR to remove them from the new testing class.